src/main.rs

```diff
@@ -100,6 +114,62 @@ fn get_username(
         })
 }
 
+/// Cleans the provisioning state marker for the current VM.
+///
+/// This removes the `.provisioned` file named after the VM ID from the
+/// configured azure-init data directory (typically `/var/lib/azure-init`).
+/// Removing this file allows provisioning to run again on the next startup.
+#[instrument]
+fn clear_provisioning_status(
```

Comment:
I'd recommend updating the function names to use `clean` instead of `clear` for consistency